File: AndroidXPatternsApp/jmvvm-app/src/main/java/com/woowrale/jmvvm/ui/search/home/SearchActivity.java
package com.woowrale.jmvvm.ui.search.home;

import android.graphics.Color;
import android.os.Build;
import android.os.Bundle;
import android.view.View;

import androidx.appcompat.widget.Toolbar;

import com.woowrale.jmvvm.R;
import com.woowrale.jmvvm.ui.base.BaseActivity;
import com.woowrale.jmvvm.ui.search.local.LocalSearchActivity;
import com.woowrale.jmvvm.ui.search.remote.RemoteSearchActivity;

import javax.inject.Inject;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class SearchActivity extends BaseActivity {

    @Inject
    SearchViewModel model;

    @BindView(R.id.toolbar)
    Toolbar toolbar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        initDagger();

        setContentView(R.layout.activity_search);
        ButterKnife.bind(this);

        setSupportActionBar(toolbar);
        whiteNotificationBar(toolbar);
    }

    protected void initDagger() {
        super.getDaggerMainComponent().inject(this);
    }

    @OnClick(R.id.btn_local_search)
    public void openLocalSearch() {
        startActivity(model.navigationTo(this, LocalSearchActivity.class).getValue());
    }

    @OnClick(R.id.btn_remote_search)
    public void openRemoteSearch() {
        startActivity(model.navigationTo(this, RemoteSearchActivity.class).getValue());
    }

    private void whiteNotificationBar(View view) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            int flags = view.getSystemUiVisibility();
            flags |= View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR;
            view.setSystemUiVisibility(flags);
            getWindow().setStatusBarColor(Color.WHITE);
        }
    }
}

File: AndroidXPatternsApp/kmvvm-app/src/main/java/com/woowrale/kmvvm/ui/search/home/SearchActivity.kt
package com.woowrale.kmvvm.ui.search.home

import android.graphics.Color
import android.os.Build
import android.os.Bundle
import android.view.View
import com.woowrale.kmvvm.R
import com.woowrale.kmvvm.ui.base.BaseActivity
import com.woowrale.kmvvm.ui.search.local.LocalSearchActivity
import com.woowrale.kmvvm.ui.search.remote.RemoteSearchActivity
import kotlinx.android.synthetic.main.activity_search.*
import kotlinx.android.synthetic.main.content_search.*
import javax.inject.Inject

class SearchActivity : BaseActivity() {

    @Inject
    lateinit var model: SearchViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        initDagger()

        setContentView(R.layout.activity_search)

        setSupportActionBar(toolbar)
        whiteNotificationBar(toolbar)

        btnLocalSearch.setOnClickListener { view ->
            startActivity(model.navigationTo(this, LocalSearchActivity::class.java).value)
        }

        btnRemoteSearch.setOnClickListener { view ->
            startActivity(model.navigationTo(this, RemoteSearchActivity::class.java).value)
        }
    }

    override fun initDagger() {
        getDaggerMainComponent().inject(this)
    }

    private fun whiteNotificationBar(view: View?) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            var flags = view!!.systemUiVisibility
            flags = flags or View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR
            view.systemUiVisibility = flags
            getWindow().setStatusBarColor(Color.WHITE)
        }
    }
}